skimage/transform/_thin_plate_splines.py

```diff
@@ -0,0 +1,316 @@
+            )
+
+        self.src = src
+        n, d = src.shape
+
+        dist = sp.spatial.distance.cdist(self.src, self.src)
+        K = _radial_basis_kernel(dist)
+        P = np.hstack([np.ones((n, 1)), src])
+        L = np.zeros((n + 3, n + 3), dtype=np.float32)
+        L[:n, :n] = K
+        L[:n, -3:] = P
+        L[-3:, :n] = P.T
+        V = np.concatenate([dst, np.zeros((d + 1, d))])
+        self.spline_mappings = np.dot(np.linalg.inv(L), V)
+        return True
+
+    def _radial_distance(self, x, y):
+        """Compute the radial distance."""
+        Pi_x = self.src[:, 0]
+        Pi_y = self.src[:, 1]
+
+        dx = Pi_x[:, np.newaxis] - x
+        dy = Pi_y[:, np.newaxis] - y
+        r = np.sqrt(dx**2 + dy**2)
+        radial_dist = _radial_basis_kernel(r)
+        return radial_dist
+
+    def _spline_function(self, x, y, radial_dist, coeffs):
+        """Solve the spline function in the X and Y directions"""
+        w = coeffs[:-3]
+        a1, ax, ay = coeffs[-3:]
+        summation = np.dot(w, radial_dist)
+        return a1 + ax * x + ay * y + summation
+
+
+def _radial_basis_kernel(r):
+    """Compute basis kernel for thin-plate splines.
+
+    Parameters
+    ----------
+    r : (4, N) ndarray
+        Input array representing the norm distance between interlandmark
+        distances for the source form and based on the (x,y) coordinates
+        for each of these points.
+
+    Returns
+    -------
+    U : (4, N) ndarray
+        Calculated kernel function U.
+    """
+    _small = 1e-8  # Small value to avoid divide-by-zero
+    r_sq = r**2
+    U = np.where(r == 0.0, 0.0, r_sq * np.log(r_sq + _small))
+    return U
+
+
+def tps_warp(
+    image, src, dst, *, output_region=None, interpolation_order=1, grid_scaling=1
+):
+    """Return an array of warped images.
+
+    Define a thin-plate-spline warping transform that warps from the
+    src to the dst, and then warp the given images by
+    that transform.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image.
+    src : (N, 2)
+        Control points at source coordinates.
+    dst : (N, 2)
+        Control points at target coordinates.
+    output_region : tuple of integers, optional
+        The region ``(xmin, ymin, xmax, ymax)`` of the output
+        image that should be produced. (Note: The region is inclusive, i.e.
+        xmin <= x <= xmax)
+    interpolation_order : int, optional
+        If 1, use linear interpolation, otherwise use
+        nearest-neighbor interpolation.
+    grid_scaling : int, optional
+        If grid_scaling is greater than 1, say x, then the transform is
+        defined on a grid x times smaller than the output image region.
+        Then the transform is bilinearly interpolated to the larger region.
+        This is fairly accurate for values up to 10 or so. Must be equal to or
+        greater than 1.
+
+    Returns
+    -------
+    warped : array_like
```

Comment:
```suggestion
    warped : ndarray
```
Right? I'd be surprised if the return type could vary...